### PR TITLE
projects:ad719x_iio:Adding STM32 Support

### PR DESCRIPTION
**Pull request by D-Disha**
1. Added STM32 Platform Support to AD719x family.
2. Updated documentation to include stm32 support

### CODE REVIEW

**Comment by D-Disha** [comment]
Hi all, are there any comments on this PR?

**Comment by D-Disha** [comment]
@jansunil @SaikiranGudla Does this look good to merge?

File: doc/sphinx/source/projects/ad719x_iio/ad719x_iio.rst [hunk 1]
```diff
@@ -52,11 +57,11 @@ to communicates with IIO device.
 Hardware Connections
 ====================
 
-Required: SDP-K1 (or alternative Mbed enabled controller board), EVAL-AD719x board
+Required: SDP-K1 (or alternative Mbed enabled controller board, or an STM32 board), EVAL-AD719x board
 and USB cable.
 
 Connect the EVAL-AD719x board to SDP-K1 board (or any other Mbed enabled controller 
-board) using jumper wires. Connect SDP-K1 board to the PC using the USB cable. 
+board an equivalent STM32 board) using jumper wires. Connect SDP-K1 board to the PC using the USB cable.
```
Comment (SaikiranGudla) [inline doc/sphinx/source/projects/ad719x_iio/ad719x_iio.rst:64]: Can add `or`  before `an ..`

File: projects/ad719x_iio/app/app_config.h [hunk 2]
```diff
@@ -23,6 +23,7 @@
 /******************************************************************************/
 /* List of supported platforms*/
 #define	MBED_PLATFORM		1
+#define STM32_PLATFORM      2
```
Comment (SaikiranGudla) [inline projects/ad719x_iio/app/app_config.h:26]: nit: Can add extra space to align with the above macro value

File: projects/ad719x_iio/app/app_config.c [hunk 3]
```diff
@@ -55,10 +59,12 @@ static struct no_os_callback_desc ext_int_callback_desc = {
 
 /* I2C init parameters */
 static struct no_os_i2c_init_param no_os_i2c_init_params = {
-	.device_id = 0,
+	.device_id = I2C_DEVICE_ID,
 	.platform_ops = &i2c_ops,
 	.max_speed_hz = 100000,
+#if(ACTIVE_PLATFORM == MBED_PLATFORM)
```
Comment (SaikiranGudla) [inline projects/ad719x_iio/app/app_config.c:65]: nit: Shall we define this for STM32 platform as well and remove this macro check similar to other projects?

**Review by jansunil — APPROVED** [review]

**Review by SaikiranGudla — APPROVED** [review]

**Pull request merged**